packages/connection/src/class-manager.php

```diff
@@ -517,13 +517,13 @@ private function internal_verify_xml_rpc_signature() {
 	}
 
 	/**
-	 * Returns true if the current site is connected to WordPress.com.
+	 * Returns true if the current site is connected to WordPress.com and has the minimum requirements to enable Jetpack UI.
 	 *
 	 * @return Boolean is the site connected?
 	 */
 	public function is_active() {
 		if ( ( new Status() )->is_no_user_testing_mode() ) {
-			return $this->is_registered();
+			return $this->is_connected();
 		}
 		return (bool) $this->get_access_token( self::CONNECTION_OWNER );
```

Comment:
We need to be careful here when it comes to calling `get_access_token`
So far the method has a `suppress_errors` arg by default set to `true`.
What worries me is that if something in that logic changes and the method returns a `WP_Error` our `bool` cast will return `true`.
I guess it wouldn't hurt to add an extra check.

Reply:
Well, this will happen only if someone decides to break backward compatibility. I think as long as we have tests that will break if that happens we are covered... don't you think?

Reply:
I confirm that the following tests currently fail when changing :
```
public function get_access_token( $user_id = false, $token_key = false, $suppress_errors = true ) {
```

to 

```
public function get_access_token( $user_id = false, $token_key = false, $suppress_errors = false ) {
```

1) Automattic\Jetpack\Connection\Test_REST_Endpoints::test_remote_authorize
Failed asserting that 500 matches expected 200.

2) Automattic\Jetpack\Connection\Test_REST_Endpoints::test_connection
Failed asserting that true is false.

3) Automattic\Jetpack\Connection\ManagerIntegrationTest::test_is_connected with data set "no blog token, blog id" (false, 1234, false)
Failed asserting that true matches expected false.

So I guess we'll be fine :)

Reply:
I added a more specifict test in https://github.com/Automattic/jetpack/pull/17832/commits/e4c59f6edab3e4c0634fbff9cafcd25e215b189a